File: app/src/main/java/com/bracketcove/postrainer/alarmreceiver/AlarmReceiverContract.java
package com.bracketcove.postrainer.alarmreceiver;

import com.bracketcove.postrainer.base.BasePresenter;
import com.bracketcove.postrainer.base.BaseView;
import com.bracketcove.postrainer.data.viewmodel.Alarm;

/**
 * Created by Ryan on 05/03/2017.
 */

public interface AlarmReceiverContract {
    interface View extends BaseView {
            String getAlarmId();

            Alarm getViewModel();

            void finishActivity();
    }

    interface Presenter extends BasePresenter {
        void onAlarmDismissClick();

    }
}

File: app/src/main/java/com/bracketcove/postrainer/alarmreceiver/AlarmReceiverContract.kt
package com.bracketcove.postrainer.alarmreceiver

/**
 * Created by Ryan on 05/03/2017.
 */

interface AlarmReceiverContract {
    interface View {
        fun finish()
    }
}

sealed class AlarmReceiverEvent {
    data class OnStart(val alarmId: String): AlarmReceiverEvent()
    data class OnDismissed(val alarmId: String): AlarmReceiverEvent()
}